packets.go

```diff
@@ -700,10 +700,11 @@ func (rows *textRows) readRow(dest []driver.Value) error {
 	if data[0] == iEOF && len(data) == 5 {
 		// server_status [2 bytes]
 		rows.mc.status = readStatus(data[3:])
-		if err := rows.mc.discardResults(); err != nil {
+		err := rows.mc.discardResults()
+		rows.mc = nil
```

Comment:
How about `rows.mc.Close()` before `rows.mc = nil`?
When discardResults() returned any error, I don't want to reuse the connection anymore.